src/node_trace_events.cc

```diff
@@ -132,6 +134,62 @@ static void CategoryGroupEnabled(const FunctionCallbackInfo<Value>& args) {
   args.GetReturnValue().Set(*category_group_enabled > 0);
 }
 
+static void EnableTracingCategories(const FunctionCallbackInfo<Value>& args) {
+#if NODE_USE_V8_PLATFORM
+  Environment* env = Environment::GetCurrent(args);
+  CHECK(args[0]->IsArray());
+  Local<Array> categories = args[0].As<Array>();
+  size_t length = categories->Length();
+  if (length == 0)
+    return;
+  std::vector<std::string> cats;
+  for (size_t n = 0; n < length; n++) {
+    Utf8Value val(env->isolate(),
+                  categories->Get(env->context(), n).ToLocalChecked());
+    cats.push_back(*val);
+  }
+  EnableTracingCategories(cats);
+#elif
+  UNREACHABLE();
+#endif
+}
+
+static void DisableTracingCategories(const FunctionCallbackInfo<Value>& args) {
+#if NODE_USE_V8_PLATFORM
+  Environment* env = Environment::GetCurrent(args);
+  CHECK(args[0]->IsArray());
+  Local<Array> categories = args[0].As<Array>();
+  size_t length = categories->Length();
+  if (length == 0)
+    return;
+  std::vector<std::string> cats;
+  for (size_t n = 0; n < length; n++) {
+    Utf8Value val(env->isolate(),
+                  categories->Get(env->context(), n).ToLocalChecked());
+    cats.push_back(*val);
+  }
+  DisableTracingCategories(cats);
+#elif
+  UNREACHABLE();
+#endif
+}
+
+static void GetTracingCategories(const FunctionCallbackInfo<Value>& args) {
+#if NODE_USE_V8_PLATFORM
+  Environment* env = Environment::GetCurrent(args);
+  auto categories = GetEnabledTracingCategories();
+  Local<Array> ret = Array::New(env->isolate(), categories.size());
+  size_t n = 0;
+  for (auto it = categories.begin(); it != categories.end(); it++) {
```

Comment:
`for (const std::string& category : categories)`? If it fits into one line you can even replace `categories` with `GetEnabledTracingCategories()` in that scenario